### PR TITLE
fix(css): parse css selectors with escape sequences (for real)

### PR DESCRIPTION
**Pull request by RedHatter**
Currently NativeScript doesn't correctly parse escape sequences in CSS selectors. There was an issue and PR (#7688, #7689) about this problem previously but only escaping a slash was implemented. My PR fully supports escape sequence from the [spec](https://www.w3.org/TR/CSS21/syndata.html#characters). This includes escaping any character or Unicode sequence.

## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [x] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

### CODE REVIEW

**Comment by vtrifonov** [comment]
teest

**Comment by vtrifonov** [comment]
test

File: nativescript-core/css/parser.ts [hunk 1]
```diff
@@ -804,10 +804,16 @@ export function parseUniversalSelector(text: string, start: number = 0): Parsed<
     return { start, end, value: { type: "*" } };
 }
 
-const simpleIdentifierSelectorRegEx = /(#|\.|:|\b)([_-\w][_-\w\d\\/]*)/gy;
+const simpleIdentifierSelectorRegEx = /(#|\.|:|\b)((?:[\w_-]|\\.)(?:[\w\d_-]|\\.)*)/gyu;
+const unicodeEscapeRegEx = /\\([0-9a-fA-F]{1,5}\s|[0-9a-fA-F]{6})/g
```
Comment (vtrifonov) [inline nativescript-core/css/parser.ts:808]: You need to add a semicolumn at the end of the row here.

**Comment by bundyo** [comment]
LGTM :+1: (after adding the semicolon)

**Comment by vtrifonov** [comment]
test

**Comment by vtrifonov** [comment]
test

**Comment by rigor789** [comment]
@RedHatter I just spent the last hour or so debugging the `.foo\:bar` selector. Traced it back to the `simpleIdentifierSelectorRegEx` and found your PR! Thanks for fixing this - my patch was definitely a temporary workaround.

**Pull request merged**